Bugsnag/Storage/BSGFileLocations.h

```diff
@@ -49,9 +50,39 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (readonly, nonatomic) NSString *systemState;
 
+/**
+ Initialize the file locations.
+ @param subdirectory If nil, use the regular shared directory. Else use the subdirectory with the given name, inside the `atomicDirectoryContainer`.
+ */
+- (instancetype) initWithSubdirectory:(NSString * _Nullable)subdirectory;
+
+/**
+ Get the singleton, initializing it with the given subdirectory. Note that it is an error to call this multiple times with differing values.
```

Comment:
This of course is quite bad design. However, bugsnag is relying on this singleton everywhere. I think it is the least bad option we currently have. At least we warn if we catch this being called with differing values.